client/state/data-layer/wpcom/sites/settings/index.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SITE_SETTINGS_SAVE,
+	SITE_SETTINGS_SAVE_FAILURE,
+	SITE_SETTINGS_SAVE_SUCCESS,
+	SITE_SETTINGS_REQUEST,
+	SITE_SETTINGS_REQUEST_FAILURE,
+	SITE_SETTINGS_REQUEST_SUCCESS,
+} from 'state/action-types';
+import { receiveSiteSettings, updateSiteSettings } from 'state/site-settings/actions';
+import { normalizeSettings } from 'state/site-settings/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+
+export const fetchSiteSettings = ( { dispatch }, action ) => {
+	const { siteId } = action;
+
+	dispatch(
+		http(
+			{
+				method: 'GET',
+				path: `/sites/${ siteId }/settings`,
+				apiVersion: '1.1',
+			},
+			action
+		)
+	);
+};
+
+export const addSiteSettings = ( { dispatch }, action, next, { name, description, settings } ) => {
+	const { siteId } = action;
+	const savedSettings = {
+		...normalizeSettings( settings ),
+		blogname: name,
+		blogdescription: description,
+	};
```

Comment:
check out here how we _assume_ that the API response gave us a proper response. also, _what_ does a proper response look like? I would like to strongly encourage us to create a sort of mapping between API data and Calypso data and validation. @samouri has been thinking about how to make this easier, but in the meantime, could you consider a `fromApi( apiResponseData )` function which explicitly shows which values it pulls out? what does `settings` look like?